Volunti/src/main/java/com/codecool/volunti/controller/RegistrationController.java

```diff
@@ -0,0 +1,154 @@
+package com.codecool.volunti.controller;
+
+
+import com.codecool.volunti.model.Organisation;
+import com.codecool.volunti.model.User;
+import com.codecool.volunti.repository.OrganisationRepository;
+import com.codecool.volunti.service.*;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.*;
+import sun.reflect.generics.reflectiveObjects.NotImplementedException;
+
+
+import javax.servlet.http.HttpSession;
+import java.util.HashMap;
+import java.util.UUID;
+
+@Controller
+@SessionAttributes({"organisation", "user"})
+public class RegistrationController {
+
+    private Logger LOGGER = LoggerFactory.getLogger(RegistrationController.class);
+    private static EmailType EMAILTYPE = EmailType.CONFIRMATION;
+
+    private OrganisationRepository organisationRepository;
+    private OrganisationService organisationService;
+    private UserService userService;
+    private EmailService emailService;
+    private ValidationService validationService = new ValidationService(organisationService, userService);
+
+
+    @Autowired
+    public RegistrationController(OrganisationRepository organisationRepository,
+                                  OrganisationService organisationService,
+                                  UserService userService,
+                                  EmailService emailService,
+                                  ValidationService validationService) {
+        this.organisationRepository = organisationRepository;
+        this.organisationService = organisationService;
+        this.userService = userService;
+        this.emailService = emailService;
+        this.validationService = validationService;
+    }
+
+
+    //render organisation registration
```

Comment:
I am a bit concerned about commenting the methods. (I know that we were already talk about comments... )